### PR TITLE
cis fixes

### PR DESCRIPTION
**Pull request by leodotcloud**
- Add benchmark version field to the report (https://github.com/rancher/rancher/issues/24562)
- Add ability to override benchmark version (https://github.com/rancher/rancher/issues/24563)
- Add support to specify skip for different benchmark versions (https://github.com/rancher/rancher/issues/23905)
- Fixed failure when the configmap for skip is malformed (https://github.com/rancher/rancher/issues/24539)
- Fixed issue when skip from API is not working when empty (https://github.com/rancher/rancher/issues/24444)
- Fixed issue with mixed result not showing nodes (https://github.com/rancher/rancher/issues/24540, https://github.com/rancher/rancher/issues/24431)

### CODE REVIEW

File: pkg/api/customization/clusterscan/clusterscan.go [hunk 1]
```diff
@@ -22,10 +22,18 @@ const (
...
 	if err := apiContext.AccessControl.CanDo(mgmtv3.ClusterScanGroupVersionKind.Group, mgmtv3.ClusterScanResource.Name, "read", apiContext, resource.Values, apiContext.Schema); err == nil {
 		status := resource.Values["status"].(map[string]interface{})
+		failed := false
+		completed := false
 		for _, cond := range convert.ToMapSlice(status["conditions"]) {
 			if cond["type"] == string(mgmtv3.ClusterScanConditionCompleted) && cond["status"] == "True" {
-				resource.Links["report"] = apiContext.URLBuilder.Link("report", resource)
+				completed = true
 			}
+			if cond["type"] == string(mgmtv3.ClusterScanConditionFailed) && cond["status"] == "True" {
```
Comment (alena1108) [inline pkg/api/customization/clusterscan/clusterscan.go]: I believe we have a norman library call for condition verification

File: pkg/controllers/user/cis/clusterScanHandler.go [hunk 2]
```diff
@@ -53,23 +63,54 @@ func (csh *cisScanHandler) Create(cs *v3.ClusterScan) (runtime.Object, error) {
...
+					Kind: "ConfigMap",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: getOverrideConfigMapName(cs),
+				},
+				Data: map[string]string{
+					ConfigFileName: cs.Spec.ScanConfig.CisScanConfig.Skip,
+				},
+			}
+			logrus.Infof("cisScanHandler: Create: creating skip override configmap %v with contents: %v", cm.Name, cs.Spec.ScanConfig.CisScanConfig.Skip)
```
Comment (alena1108) [inline pkg/controllers/user/cis/clusterScanHandler.go]: is it safe to print the map contents in info? Can there be any sensitive information?

Comment (leodotcloud) [reply pkg/controllers/user/cis/clusterScanHandler.go]: nope, there is no sensitive info. Only the test numbers which are being skipped.

File: pkg/api/customization/clusterscan/clusterscan.go (hunk 1 above)
Comment (leodotcloud) [reply pkg/api/customization/clusterscan/clusterscan.go]: I think I had used etcdbackup code as reference. https://github.com/rancher/rancher/blob/master/pkg/api/customization/etcdbackup/formatter.go#L14

**Comment by alena1108** [comment]
TF is unrelated

**Pull request merged**